src/managers/connect.py

```diff
@@ -273,21 +273,18 @@ def stop_connector(self, relation_id: int) -> None:
         wait=wait_fixed(3),
         stop=stop_after_attempt(5),
         retry=retry_any(
-            retry_if_result(lambda result: result is False), retry_if_exception(lambda _: True)
+            retry_if_result(lambda result: not result), retry_if_exception(lambda _: True)
         ),
-        retry_error_callback=lambda _: False,
+        retry_error_callback=lambda _: HealthResponse(status_code=-1),
     )
-    def health_check(self) -> bool:
+    def health(self) -> HealthResponse:
```

Comment:
Small nit, but a bit weird to not call it `healthy`, since the calls are like `if health`

Reply:
Done https://github.com/canonical/kafka-connect-operator/pull/30/commits/0e86062c6aa26ff1f94d707ad68afb03faa6c577